components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/CertificateMgtUtils.java

```diff
@@ -185,12 +192,144 @@ public ResponseCode removeCertificateFromTrustStore(String alias) {
+
+            if (x509Certificate.getNotAfter().getTime() <= System.currentTimeMillis()) {
+                log.error("Could not update the certificate. The certificate expired.");
+                return ResponseCode.CERTIFICATE_EXPIRED;
+            }
+            // If the certificate is not expired, delete the existing certificate and add the new cert.
+            trustStore.deleteEntry(alias);
+            //Store the certificate in the trust store.
+            trustStore.setCertificateEntry(alias, newCertificate);
+            fileOutputStream = new FileOutputStream(trustStoreFile);
+            trustStore.store(fileOutputStream, TRUST_STORE_PASSWORD);
+        } catch (IOException e) {
+            throw new CertificateManagementException("Error updating certificate.", e);
+        } catch (CertificateException e) {
+            throw new CertificateManagementException("Error generating the certificate.", e);
+        } catch (NoSuchAlgorithmException e) {
+            throw new CertificateManagementException("Error loading the keystore.", e);
+        } catch (KeyStoreException e) {
+            throw new CertificateManagementException("Error updating the certificate in the keystore.", e);
+        } finally {
+            closeStreams(fileOutputStream, certificateStream, localTrustStoreStream);
+        }
+        return ResponseCode.SUCCESS;
+    }
+
+    /**
+     * Method to get the information of the certificate.
+     *
+     * @param alias : Alias of the certificate which information should be retrieved
+     * @return : The details of the certificate as a MAP.
+     */
+    public CertificateInformationDTO getCertificateInformation(String alias) throws CertificateManagementException {
+
+        CertificateInformationDTO certificateInformation = new CertificateInformationDTO();
+        File trustStoreFile = new File(TRUST_STORE);
+        try {
+            localTrustStoreStream = new FileInputStream(trustStoreFile);
+            KeyStore trustStore = KeyStore.getInstance(KeyStore.getDefaultType());
+            trustStore.load(localTrustStoreStream, TRUST_STORE_PASSWORD);
+
+            if (trustStore.containsAlias(alias)) {
+                X509Certificate certificate = (X509Certificate) trustStore.getCertificate(alias);
+                certificateInformation.setStatus(certificate.getNotAfter().getTime() > System.currentTimeMillis() ?
+                        "Active" : "Expired");
+                certificateInformation.setFrom(certificate.getNotBefore().toString());
+                certificateInformation.setTo(certificate.getNotAfter().toString());
+                certificateInformation.setSubject(certificate.getSubjectDN().toString());
+                certificateInformation.setVersion(String.valueOf(certificate.getVersion()));
+            }
+        } catch (IOException e) {
+            throw new CertificateManagementException("Error in loading the certificate.", e);
```

Comment:
Error message should be including context related information, in this case shall we add alias